src/components/utils/src/threads/thread_posix.cc

```diff
@@ -261,15 +270,27 @@ void Thread::stop() {
   SDL_DEBUG("Stopped thread #" << handle_ << " \"" << name_ << " \"");
 }
 
-void Thread::join() {
+void Thread::join(const bool force_stop) {
   SDL_AUTO_TRACE();
   DCHECK_OR_RETURN_VOID(!IsCurrentThread());
 
-  stop();
+  if (force_stop) {
+    stop();
+  }
 
```

Comment:
@orhan-mehmedov BTW seems like here somebody could call start and `join(force)` will not stop the thread. I mean that `stop()` and `join()` should be called under one lock